File: ruby/lib/enolib/elements/fieldset_entry.rb
# frozen_string_literal: true

module Enolib
  class FieldsetEntry < ValueElementBase
    attr_reader :instruction # TODO: Revisit this hacky exposition
    
    def parent
      @parent || Fieldset.new(@context, @instruction[:parent])
    end

    def to_s
      "#<Enolib::FieldsetEntry key=#{@instruction[:key]} value=#{print_value}>"
    end
  end
end

File: python/enolib/elements/fieldset_entry.py
from . import fieldset
from .value_element_base import ValueElementBase


class FieldsetEntry(ValueElementBase):
    def __repr__(self):
        return f"<class FieldsetEntry key={self._instruction['key']} value={self._print_value()}>"

    def parent(self):
        return self._parent or fieldset.Fieldset(self._context, self._instruction.parent)
